src/Geminals/AP1roGGeminalCoefficients.cpp

```diff
@@ -69,32 +97,76 @@ AP1roGGeminalCoefficients::AP1roGGeminalCoefficients(size_t N_P, size_t K) :
  *
  *  @return the AP1roG geminal coefficients in the weak interaction limit
  */
-AP1roGGeminalCoefficients AP1roGGeminalCoefficients::WeakInteractionLimit(const SQHamiltonian<double>& sq_hamiltonian, size_t N_P) {
+AP1roGGeminalCoefficients AP1roGGeminalCoefficients::WeakInteractionLimit(const SQHamiltonian<double>& sq_hamiltonian, const size_t N_P) {
 
     const auto K = sq_hamiltonian.dimension();
     const auto number_of_geminal_coefficients = AP1roGGeminalCoefficients::numberOfGeminalCoefficients(N_P, K);
     const auto& h = sq_hamiltonian.core().parameters();  // core Hamiltonian integrals
     const auto& g = sq_hamiltonian.twoElectron().parameters();  // two-electron integrals
 
     // Provide the weak interaction limit values for the geminal coefficients
-    VectorX<double> g_vector = VectorX<double>::Zero(number_of_geminal_coefficients);
-    for (size_t mu = 0; mu < number_of_geminal_coefficients; mu++) {
-        size_t i = GQCP::matrixIndexMajor(mu, K, N_P);
-        size_t a = GQCP::matrixIndexMinor(mu, K, N_P);
-
-        g_vector(mu) = - g(a,i,a,i) / (2 * (h(a,a) - h(i,i)));
+    BlockMatrix<double> G (0, N_P, N_P, K);
+    for (size_t i = 0; i < N_P; i++) {
+        for (size_t a = N_P; a < K; a++) {
+            G(i,a) = - g(a,i,a,i) / (2 * (h(a,a) - h(i,i)));
+        }
```

Comment:
For the reviewers: I think this is an excellent example of the cleaner code when using `BlockMatrix`.